packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -32,9 +168,97 @@ export default class TabGeneral extends Component {
     }
   }
 
-  render() {
-    const { sr, vdis, vdiSnapshots, unmanagedVdis } = this.props
+  _getDiskGroups = createSelector(
+    () => this.props.vdis,
+    () => this.props.vdiSnapshots,
+    () => this.props.unmanagedVdis,
+    (vdis, vdiSnapshots, unmanagedVdis) => {
+      const diskParents = []
+      const groups = []
+      const snapshotsByVdi = groupBy(vdiSnapshots, '$snapshot_of')
+      const unmanagedVdisById = keyBy(unmanagedVdis, 'id')
+      let baseCopiesUsage
+      let bc
+      let nBaseCopies
+      let root
+      // group VDIs by their initial base copy.
+      const disksByRoot = groupBy(
+        vdis.map(({ id, parent, snapshots, ...vdi }) => {
+          baseCopiesUsage = 0
+          nBaseCopies = 0
+          root = undefined
+          while ((bc = unmanagedVdisById[parent]) !== undefined) {
+            root = bc.id
+            parent = bc.parent
+            if (!diskParents.includes(root)) {
+              nBaseCopies++
+              baseCopiesUsage += bc.usage
+              diskParents.push(root)
+            }
+          }
+
+          if ((snapshots = snapshotsByVdi[id]) !== undefined) {
+            // snapshot can have base copy without active VDI
+            snapshots.forEach(({ parent }) => {
+              if (parent !== undefined && !diskParents.includes(parent)) {
+                nBaseCopies++
+                baseCopiesUsage += unmanagedVdisById[parent].usage
+                diskParents.push(parent)
+              }
+            })
+          }
+
+          return {
+            ...vdi,
+            id,
+            baseCopiesUsage,
+            nBaseCopies,
+            root: root || id,
+            snapshots: snapshots || [],
+          }
+        }),
+        'root'
+      )
 
+      let orphanedVdiSnapshots
+      if ((orphanedVdiSnapshots = snapshotsByVdi[undefined]) !== undefined) {
+        orphanedVdiSnapshots.forEach(snapshot => {
+          groups.push(snapshot)
+        })
+      }
+
+      let snapshots
+      let usage
+      // group collection of VDIs and their snapshots and base copies.
+      forEach(disksByRoot, vdis => {
+        baseCopiesUsage = 0
+        nBaseCopies = 0
+        snapshots = []
+        usage = 0
+        vdis.forEach(vdi => {
+          usage += vdi.usage
+          nBaseCopies += vdi.nBaseCopies
```

Comment:
I don't get it. VDIs attached to the same base copy root will have some common base copies, right? So you can't just add them.